eng/testing/AppleRunnerTemplate.sh

```diff
@@ -39,14 +36,29 @@ while true; do
     fi
 done
 
-XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
+export XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
```

Comment:
The output dir should be `$HELIX_WORKITEM_UPLOAD_ROOT` for Helix to upload logs automatically